source/h1_stream.c

```diff
@@ -217,9 +286,79 @@ struct aws_h1_stream *aws_h1_stream_new_request_handler(const struct aws_http_re
      * with the lock here. */
     stream->base.id = aws_http_connection_get_next_stream_id(options->server_connection);
 
+    /* Request-handler (server) streams don't need user to call activate() on them.
+     * Since these these streams can only be created on the event-loop thread,
+     * it's not possible for callbacks to fire before the stream pointer is returned.
+     * (Clients must call stream.activate() because they might create a stream on any thread) */
+    stream->synced_data.api_state = AWS_H1_STREAM_API_STATE_ACTIVE;
+
     stream->base.server_data = &stream->base.client_or_server_data.server;
     stream->base.server_data->on_request_done = options->on_request_done;
     aws_atomic_fetch_add(&stream->base.refcount, 1);
 
     return stream;
 }
+
+int aws_h1_stream_send_response(struct aws_h1_stream *stream, struct aws_http_message *response) {
+    struct aws_h1_connection *connection = s_get_h1_connection(stream);
```

Comment:
This implementation was moved from h1_connection.c to h1_stream.c,
seems like it belongs here more
and h1_connection.c is nearly 2000 lines long, it certainly doesn't need any more code than strictly necessary